mysql-plugin/src/main/java/io/cdap/plugin/mysql/MysqlUtil.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.mysql;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+
+/**
+ * MySQL util methods.
+ */
+public final class MysqlUtil {
+  private MysqlUtil() {
+    throw new AssertionError("Should not instantiate static utility class.");
+  }
+
+  /**
+   * Composes immutable map of the MySQL specific arguments.
+   *
+   * @param autoReconnect                     should the driver try to re-establish stale and/or dead connections.
+   * @param useCompression                    specifies if compression must be enabled.
+   * @param useSSL                            specifies if SSL encryption must be turned on.
+   * @param clientCertificateKeyStoreUrl      URL of the client certificate KeyStore.
+   * @param clientCertificateKeyStorePassword password for the client certificates KeyStore.
+   * @param trustCertificateKeyStoreUrl       URL of the trusted root certificate KeyStore.
+   * @param trustCertificateKeyStorePassword  password for the trusted root certificates KeyStore.
+   * @return immutable map of the MySQL specific arguments
+   */
+  public static Map<String, String> composeDbSpecificArgumentsMap(Boolean autoReconnect,
```

Comment:
Is it possible to have a base MysqlConf class that has all of these properties and move this logic to that class?

Reply:
There is no straightforward way to do so due to the existing hierarchy of config classes.